src/Wrappers.dfy

```diff
@@ -69,4 +69,26 @@ module Wrappers {
       value
     }
   }
+
+  datatype Outcome<E> = Pass | Fail(error: E)
```

Comment:
@robin-aws What about calling this type `Check`?  A check can pass or fail. And if a check passes then there is no additional information needed. It is only when a check fails that we want to know things.

Reply:
I think that would make less sense, since this is the result of a check rather than the act of checking itself. `Check` would fit better as an alternative to `Need` instead, but I find `Need` fits into the idiom of verbs already in Dafny better so I'm happier with `Need`.

Reply:
Yup, just options.